dvc/output/base.py

```diff
@@ -297,8 +297,9 @@ def checkout(self, force=False, progress_callback=None, tag=None):
             progress_callback=progress_callback,
         )
 
-    def remove(self, ignore_remove=False):
-        self.remote.remove(self.path_info)
+    def remove(self, ignore_remove=False, force=True):
+        self.cache.safe_remove(self.path_info, force=force)
```

Comment:
And what if output doesn't use cache? I'm pretty sure you've mentioned this :)

Reply:
I did, but removed it since the tests passed :sweat_smile: 
I still think it make sense, I'll add it.